File: app/src/main/java/cn/ifafu/ifafu/mvp/web/WebContract.java
package cn.ifafu.ifafu.mvp.web;

import cn.ifafu.ifafu.mvp.base.i.IZFModel;
import cn.ifafu.ifafu.mvp.base.i.IZFPresenter;
import cn.ifafu.ifafu.mvp.base.i.IView;
import io.reactivex.Observable;

class WebContract {
    interface Presenter extends IZFPresenter {

    }

    interface View extends IView {
        void loadUrl(String url);

        void setTitle(String title);
    }

    interface Model extends IZFModel {
        /**
         * 获取首页Url
         *
         * @return 首页Url
         */
        String getMainUrl();

        Observable<String> getMainHtml();
    }

}

File: app/src/main/java/cn/ifafu/ifafu/mvp/web/WebContract.kt
package cn.ifafu.ifafu.mvp.web

import cn.ifafu.ifafu.base.mvp.IModel
import cn.ifafu.ifafu.base.mvp.IPresenter
import cn.ifafu.ifafu.base.mvp.IView
import io.reactivex.Observable

class WebContract {
    interface Presenter: IPresenter

    interface View : IView {
        fun loadUrl(url: String)

        fun setTitle(title: String? = null, resId: Int? = null)
    }

    interface Model: IModel {
        /**
         * 获取首页Url
         *
         * @return 首页Url
         */
        fun getMainUrl(): String

        fun loadMainHtml(): Observable<MutableMap<String, String>>
    }

}
